backend/src/plugins/twitter-client/lib/utils/knowledge/knowledge-processing.ts

```diff
@@ -91,99 +94,59 @@ export async function extractAndStoreKnowledge(
-      for (let i = 0; i < entities.length; i++) {
-        const entityName = entities[i];
-
-        // Skip entities without content
-        if (!entityName) {
-          continue;
-        }
-
-        // Create a knowledge entry for each entity
-        const entityKnowledge: RAGKnowledgeItem = {
-          id: stringToUuid(uuidv4()),
-          agentId: runtime.agentId,
-          content: {
-            text: entityName,
-            metadata: {
-              source: 'twitter',
-              sourceId: tweet.tweet_id,
-              sourceUrl: tweet.permanentUrl,
-              authorUsername: tweet.username,
-              authorUserId: tweet.userId?.toString(),
-              entityName,
-              entityType: tweetType === 'factual' ? 'fact' : 'opinion',
-              isOpinion: tweetType === 'opinion',
-              isFactual: tweetType === 'factual' || tweetType === 'informative',
-              confidence: analysis.contentAnalysis.confidence || 0.5,
-              relatedTopics: topics,
-              timestamp: tweet.timestamp,
-              originalId: tweetKnowledgeId,
-              isChunk: true,
-              chunkIndex: i,
-              isShared: true, // Make entities shareable across agents
-            },
-          },
-          embedding: undefined, // Will be generated by the embedding system
-          createdAt: Date.now(),
-        };
-
-        await runtime.databaseAdapter.createKnowledge(entityKnowledge);
-      }
+    await runtime.messageManager.addEmbeddingToMemory(tweetMemory);
 
+    if (tweetMemory.embedding) {
       elizaLogger.info(
-        `${logPrefix} Stored ${entities.length} entity knowledge items for tweet ${tweet.tweet_id}`,
-      );
-    }
-
-    // If the tweet is a thread, also store a summarized version
-    if (tweet.isThreadMerged && tweet.threadSize > 1) {
-      const threadSummary: RAGKnowledgeItem = {
-        id: stringToUuid(`tweet-thread-${tweet.tweet_id}`),
-        agentId: runtime.agentId,
-        content: {
-          text: `Thread summary: ${tweet.text.substring(0, 200)}...`,
-          metadata: {
-            source: 'twitter',
-            sourceId: tweet.tweet_id,
-            sourceUrl: tweet.permanentUrl,
-            authorUsername: tweet.username,
-            authorUserId: tweet.userId?.toString(),
-            isThreadSummary: true,
-            threadSize: tweet.threadSize,
-            topics,
-            timestamp: tweet.timestamp,
-            originalId: tweetKnowledgeId,
-            isChunk: true,
-            chunkIndex: 0,
-          },
+        `${logPrefix} Successfully generated embedding for tweet ${tweet.tweet_id}`,
+        {
+          embeddingSize:
+            tweetMemory.embedding instanceof Float32Array
+              ? tweetMemory.embedding.length
+              : tweetMemory.embedding.length,
+          embeddingType:
+            tweetMemory.embedding instanceof Float32Array
+              ? 'Float32Array'
+              : 'Array',
```

Comment:
keep those long logging in debug. Use only minimal stuff and required stuff for info.